platforms/android/tangram/src/main/java/com/mapzen/tangram/MapView.java

```diff
@@ -94,20 +93,23 @@ protected MapController mapInitInBackground(@Nullable final HttpHandler handler)
                 return mapController;
             }
         }
-        return getMapInstance(handler);
+        return new MapController(this.getContext());
     }
 
     /**
      * Should be executed from the UI thread
      * @param controller {@link MapController} created in a background thread
+     * @param handler {@link HttpHandler} required for network handling
      * @param callback {@link MapReadyCallback}
      */
-    protected void onMapInitOnUIThread(@Nullable final MapController controller, @NonNull final MapReadyCallback callback) {
+    protected void onMapInitOnUIThread(@Nullable final MapController controller, @Nullable final HttpHandler handler,
+                                       @NonNull final MapReadyCallback callback) {
         synchronized (lock) {
             if (mapController == null && controller != null) {
+                glSurfaceView = createGLSurfaceView();
```

Comment:
One nitpick we discussed during our meeting, it will be good as a sanity check to make sure glSurfaceView has not been previously created, though its probably not possible.

Reply:
I'm not sure where this would helps. 

For sanity I would propose to not allow getMapAsync to be called twice since the MapController could be created once or twice in that case currently.. One moment I'll push a fix to ensure MapController can only be created once

Reply:
fb0bc17 looks good to me.

Reply:
.. no this solution is not really good either. Lets move this to another PR

Reply:
:/ what did I miss in my review!

Reply:
That change should improves things - nothing missed I hope :) To me it's not clear why we support to call getMapAsync more than once

Reply:
This process is surprisingly tricky. I think the glSurfaceView initialization has to be matched one-to-one with the map controller initialization because we can't set a renderer on the glSurfaceView more than once (GLSurfaceView throws if we try to).